McCorrections/src/EventWeights_HH.cpp

```diff
@@ -56,17 +56,23 @@ ntuple::ProdSummary EventWeights_HH::GetSummaryWithWeights(const std::shared_ptr
     static const WeightingMode shape_weights(WeightType::PileUp, WeightType::BSM_to_SM, WeightType::DY,
                                              WeightType::TTbar, WeightType::Wjets, WeightType::GenEventWeight);
     static const WeightingMode shape_weights_withTopPt = shape_weights | WeightingMode(WeightType::TopPt);
+    static const WeightingMode shape_weights_withPileUp = shape_weights | WeightingMode(WeightType::PileUp);
 
     auto summary_tuple = ntuple::CreateSummaryTuple("summary", file.get(), true, ntuple::TreeState::Full);
     auto summary = ntuple::MergeSummaryTuple(*summary_tuple);
     summary.totalShapeWeight = 0;
     summary.totalShapeWeight_withTopPt = 0;
 
     const auto mode = shape_weights & weighting_mode;
+    //TopPt
     const auto mode_withTopPt = shape_weights_withTopPt & weighting_mode;
     const bool calc_withTopPt = mode_withTopPt.count(WeightType::TopPt);
+    //PileUp
+    const auto mode_withPileUp = shape_weights_withPileUp & weighting_mode;
+    const auto mode_withoutPileUp = weighting_mode;
+    const bool calc_withPileUp = mode_withPileUp.count(WeightType::PileUp);
```

Comment:
- `mode_withPileUp` is not needed because it is equivalent to `mode`.
- `mode_withoutPileUp` should be `WeightingMode(mode).erase(WeightType::PileUp)`
- `calc_withPileUp  = mode.count(WeightType::PileUp);`